pkg/kubectl/resource/mapper.go

```diff
@@ -38,7 +38,7 @@ type Mapper struct {
 // InfoForData creates an Info object for the given data. An error is returned
 // if any of the decoding or client lookup steps fail. Name and namespace will be
 // set into Info if the mapping's MetadataAccessor can retrieve them.
-func (m *Mapper) InfoForData(data []byte, source string) (*Info, error) {
+func (m *Mapper) InfoForData(data []byte, source string, local bool) (*Info, error) {
```

Comment:
> Passing a different mapper to NewBuilder seems also work, I need to figure out how. Can you give me some hits?

Sure.  You need to change this line here:  https://github.com/kubernetes/kubernetes/pull/11368/files#diff-c1b6a3c269cbdfd64b769968834ea42eR104.  If you got `--local`, you can initialize your builder with a stub `ClientMapper` that always returns `nil` .